### ISSUE TITLE
Allow human-friendly date/time formats in input[type=date] and input[type=datetime-local]

### ISSUE THREAD

**Issue by PawelDecowski** [angular/angular.js#11837]
Currently, for example, `input[type=datetime-local]` requires the input to be in the format `yyyy-MM-ddTHH:mm:s` which in most cases is useless for end users.

The date/time `input`s should allow custom formatting (while the model remains a `Date` object).

**Comment by Narretz** [comment]
I agree that the current state is not pretty. I think this could be a good fit for a third-party module, though (implemented via parsers / formatters):
- browser support is still fragmented for the date input, so many people will still use full fledged date pickers. Then again, chrome already has a date picker
- there are many different date formats, multiplied by locales. This would be a heavy burden on the core
